rsocket-core/src/main/java/io/rsocket/metadata/MimeMetadataCodec.java

```diff
@@ -0,0 +1,145 @@
+   * @return the encoded mime
+   */
+  public static ByteBuf encode(ByteBufAllocator allocator, String mimeType) {
+    if (mimeType == null || mimeType.length() == 0) {
+      throw new IllegalArgumentException("Mime type null or length is zero");
+    }
+    WellKnownMimeType wkn = WellKnownMimeType.fromString(mimeType);
+    if (wkn == WellKnownMimeType.UNPARSEABLE_MIME_TYPE) {
+      return encodeCustomMime(allocator, mimeType);
+    } else {
+      return encodeWellKnowMime(allocator, wkn);
+    }
+  }
+
+  /**
+   * Encode multiple {@link WellKnownMimeType} or custom mime type into a newly allocated {@link
+   * ByteBuf}
+   *
+   * @param allocator the {@link ByteBufAllocator} to use to create the buffer.
+   * @param mimeTypes mime types
+   * @return the encoded mimes
+   */
+  public static ByteBuf encode(ByteBufAllocator allocator, List<String> mimeTypes) {
+    if (mimeTypes == null || mimeTypes.size() == 0) {
+      throw new IllegalArgumentException("Mime types empty");
+    }
+    CompositeByteBuf compositeMimeByteBuf = allocator.compositeBuffer();
+    for (String mimeType : mimeTypes) {
+      compositeMimeByteBuf.addComponents(true, encode(allocator, mimeType));
+    }
+    return compositeMimeByteBuf;
+  }
+
+  /**
+   * Encode a custom mime type into a newly allocated {@link ByteBuf}.
+   *
+   * <p>This larger representation encodes the mime type representation's length on a single byte,
+   * then the representation itself
+   *
+   * @param allocator the {@link ByteBufAllocator} to use to create the buffer.
+   * @param customMime a custom mime type to encode.
+   * @return the encoded mime
+   */
+  public static ByteBuf encodeCustomMime(ByteBufAllocator allocator, String customMime) {
+    ByteBuf mime = allocator.buffer(1 + customMime.length());
+    // reserve 1 byte for the customMime length
+    // /!\ careful not to read that first byte, which is random at this point
+    int writerIndexInitial = mime.writerIndex();
+    mime.writerIndex(writerIndexInitial + 1);
+
+    // write the custom mime in UTF8 but validate it is all ASCII-compatible
+    // (which produces the right result since ASCII chars are still encoded on 1 byte in UTF8)
+    int customMimeLength = ByteBufUtil.writeUtf8(mime, customMime);
+    if (!ByteBufUtil.isText(mime, mime.readerIndex() + 1, customMimeLength, CharsetUtil.US_ASCII)) {
+      mime.release();
+      throw new IllegalArgumentException("custom mime type must be US_ASCII characters only");
+    }
+    if (customMimeLength < 1 || customMimeLength > 128) {
+      mime.release();
+      throw new IllegalArgumentException(
+          "custom mime type must have a strictly positive length that fits on 7 unsigned bits, ie 1-128");
+    }
+    mime.markWriterIndex();
+    // go back to beginning and write the length
+    // encoded length is one less than actual length, since 0 is never a valid length, which gives
+    // wider representation range
+    mime.writerIndex(writerIndexInitial);
```

Comment:
Isn't `writerIndexInitial` always 0? Why not just use 0 then to make that clear.